controllers/run_controller_test.go

```diff
@@ -146,4 +162,47 @@ var _ = Describe("RunController", func() {
 			}, timeout, interval).Should(Succeed())
 		})
 	})
+	Context("Retrieve terraform state", func() {
+		It("Checks if terraform state was retrieved", func() {
+			workspace := &terraformv1.Workspace{}
+			s3Bucket := os.Getenv("SCIPIAN_STATE_BUCKET")
+			filePath := fmt.Sprintf("%s/%s/%s", "foo", "bar", core.TFStateFileName)
+			directoryPath := fmt.Sprintf("%s/%s", "foo", "bar")
+
```

Comment:
Lines 172-176 will fail sometimes, as the `k8sClient.Get` may take some time to succeed. Instead, wrap these kinds of queries in `Eventually()...` functions. Example refactor:
```go
			Eventually(func() error {
				return k8sClient.Get(ctx, workspaceKey, workspace)
			}, timeout, interval).Should(Succeed())

			Eventually(func() error {
				return k8sClient.Get(ctx, workspaceKey, foundWorkspaceJob)
			}, timeout, interval).Should(BeNil())
```

Reply:
Also note, you can use the `ctx` variable which is declared above, to avoid writing `context.TODO()` every time.